src/electionguard/ballot.py

```diff
@@ -19,15 +23,13 @@
 
 
 def _list_eq(
-    list1: Sequence[ElectionObjectBase], list2: Sequence[ElectionObjectBase]
+    list1: Sequence[OrderedObjectBase], list2: Sequence[OrderedObjectBase]
 ) -> bool:
     """
     We want to compare lists of election objects as if they're sets. We fake this by first
-    sorting them on their object ids, then using regular list comparison.
+    sorting them on their sequence order, then using regular list comparison.
     """
-    return sorted(list1, key=lambda x: x.object_id) == sorted(
-        list2, key=lambda x: x.object_id
-    )
+    return sequence_order_sort(list1) == sequence_order_sort(list2)
```

Comment:
This change could go unnoticed so calling it out. This modifies the sort of equality to rely on `sequence_order` instead of `object_id`. Considering `sequence_order` represents ordering this makes more sense and is consistent with how the items should be sorted.